src/components/binding/binding-panel.js

```diff
@@ -213,13 +187,44 @@ genotet.BindingPanel.prototype.updateTracks = function() {
       this.container.find('#genes #track-' + index).remove();
     }
   }
+
+  this.signal('loadBindingList');
+
+  this.container.find('#genes .glyphicon-remove')
+    .css('display', this.data.tracks.length == 1 ? 'none' : '');
+};
+
+/**
+ * Updates the track list for panel.
+ */
+genotet.BindingPanel.prototype.updateTrackList = function() {
```

Comment:
Well, it looks like updateTracks() is now somewhat an entry point like the update() mentioned, and updateTrackList() is almost the DOM handler that does the 2nd and 3rd steps above. 

A few more suggestions:
- Make updateTracks() a clean entry point of the entire update procedure. Remove all DOM manipulation to updateTrackList.
- Give updateTrackList a better name to indicate that it updates the DOMs of the track list. Explain in comments that updateTrackList is a process that must follow updateTracks().